services/task/task.py

```diff
@@ -34,7 +34,7 @@
 
 
 class TaskService(object):
-    def _create_signature(self, name, args=None, kwargs=None):
+    def _create_signature(self, name, args=None, kwargs=None, immutable=False):
```

Comment:
this is for my own understanding, what's the immutable for?

Reply:
Yeah, I had to look this up in the Celery docs.  When we use Celery's `chain` or `link` then the results of the previous task are passed as arguments to the next task in the chain (this is the default and is when `immutable=False`).  I left a small comment about this where I pass `immutable=True` and the goal was to prevent the results of the notify task from being passed as args to the upload task.